omc3/scripts/linfile_clean.py

```diff
@@ -0,0 +1,209 @@
+from omc3.utils import logging_tools
+from omc3.utils.iotools import PathOrStr, save_config
+from omc3.utils.outliers import get_filter_mask
+
+LOG = logging_tools.get_logger(__name__)
+
+
+def get_params():
+    return EntryPointParameters(
+        files=dict(
+            required=True,
+            type=PathOrStr,
+            nargs='+',
+            help="List of paths to the lin-files, including suffix.",
+        ),
+        # restore backup
+        restore=dict(
+            action="store_true",
+            help=("Restore the latest backup of the files. "
+                  "If this parameter is given, no cleaning is performed."),
+        ),
+        # for actual cleaning
+        columns=dict(
+            nargs='+',
+            type=str,
+            help="Columns to clean on.",
+        ),
+        limit=dict(
+            type=float,
+            help="Do not clean data-points deviating less than this limit from the average.",
+            default=0.0,
+        ),
+        backup=dict(
+            action="store_true",
+            help=("Create a backup of the files before overwriting. "
+                  "The backups are numbered, with the highest number being "
+                  "the latest backup.")
+        ),
+    )
+
+
+@entrypoint(get_params(), strict=True)
+def main(opt):
+    """Main function, to parse commandline input and separate restoration
+    from cleaning."""
+    save_config(Path('.'), opt, __file__)
+    if opt.restore:
+        restore_files(opt.files)
+        return
+
+    if opt.columns is None:
+        raise ValueError("The option 'columns' is required for cleaning.")
+    clean_columns(opt.files, opt.columns, opt.limit, opt.backup)
+
+
+# Restore ----------------------------------------------------------------------
+
+def restore_files(files: Sequence[Union[Path, str]]):
+    """Restore backupped files."""
+    failed = []
+    for file in files:
+        file = Path(file)
+        try:
+            _restore_file(file)
+        except IOError as e:
+            failed.append(str(e))
+
+    if len(failed):
+        all_errors = '\n'.join(failed)
+        if len(failed) == len(files):
+            raise IOError(f"Restoration of ALL files has failed\n{all_errors}")
+        raise IOError(f"Restoration of some files has failed, "
+                      f"but the others were restored:\n{all_errors}")
+
+    LOG.info("Restoration successfully completed.")
+
+
+def _restore_file(file):
+    counter = 1
+    backup_file = _get_backup_filepath(file, counter)
+    if not backup_file.exists():
+        raise IOError(f"No backups found for file {file.name}")
+
+    # get last existing backup file
+    while backup_file.exists():
+        counter += 1
+        backup_file = _get_backup_filepath(file, counter)
+    backup_file = _get_backup_filepath(file, counter-1)
+
+    # restore found  file
+    LOG.info(f"Restoring last backup file {backup_file.name} to {file.name}.")
```

Comment:
I feel like maybe this should be a `debug` level logging statement

Reply:
that helps you to see, which files were restored though... I think that's important. These functions are not really useful anywhere else but in this script (for now). So I think `info` is good here (thought about it for quite a while)